torch/distributed/_tensor/rng_tracker.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+from typing import Dict
+
+import torch
+import torch.distributed as dist
+
+from torch import Tensor
+
+
+class CudaRNGStatesTracker:
```

Comment:
meta comment: I think this class and `ParallelRNGStyle` should be merged, and we should put this and the global variable in `random.py` instead

Reply:
The main reason I separate them is to remove circular import dependency:

In DeviceMesh, we use RNG tracker to synchronize. But some logic in Random.py uses DeviceMesh and this causes circular dependency. 

Btw I realize that MegatronStyle can be moved to TP folder.